test/test_deployer.go

```diff
@@ -144,7 +165,55 @@ func (s *DeployerSuite) TestRollback(t *c.C) {
 	// check that the old formation is the same and there's no new formation
 	f, err := s.controllerClient(t).GetFormation(deployment.AppID, oldReleaseID)
 	t.Assert(err, c.IsNil)
-	t.Assert(f.Processes, c.DeepEquals, map[string]int{"crasher": 2})
+	t.Assert(f.Processes, c.DeepEquals, map[string]int{"failer": 2})
 	_, err = s.controllerClient(t).GetFormation(deployment.AppID, releaseID)
 	t.Assert(err, c.NotNil)
 }
+
+func (s *DeployerSuite) TestTransitioning(t *c.C) {
+	app, _ := s.createApp(t)
+	route := "ping-transitioning.dev"
+	s.createFormation(t, app, map[string]int{"ping": 1})
+	t.Assert(flynn(t, "/", "-a", app.Name, "route", "add", "http", "-s", "ping-service", route), Succeeds)
+	<-time.After(1 * time.Second) // give the job a bit of time to start
+
+	done := make(chan bool)
+
+	go func() {
+		deployment := s.createDeployment(t, app, "all-at-once")
+		events := make(chan *ct.DeploymentEvent)
+		stream, err := s.controllerClient(t).StreamDeployment(deployment.ID, events)
+		t.Assert(err, c.IsNil)
+		defer stream.Close()
+		releaseID := deployment.NewReleaseID
+		oldReleaseID := deployment.OldReleaseID
+
+		expected := []*ct.DeploymentEvent{
+			{ReleaseID: releaseID, JobType: "ping", JobState: "starting", Status: "running"},
+			{ReleaseID: releaseID, JobType: "ping", JobState: "up", Status: "running"},
+			{ReleaseID: oldReleaseID, JobType: "ping", JobState: "stopping", Status: "running"},
+			{ReleaseID: oldReleaseID, JobType: "ping", JobState: "down", Status: "running"},
+			{ReleaseID: releaseID, JobType: "", JobState: "", Status: "complete"},
+		}
+
+		defer close(done)
+		waitForDeploymentEvents(t, events, expected)
```

Comment:
As discussed, test assertions cannot be made inside a goroutine due to the way go-check works, so this will need to send errors via a channel to the main test goroutine.